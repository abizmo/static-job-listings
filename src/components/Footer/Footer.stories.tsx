import React from 'react';
import { ComponentMeta } from '@storybook/react';

import Footer from './Footer';

export default {
  title: 'Layout/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>;

export function Primary(): JSX.Element {
  return <Footer />;
}
