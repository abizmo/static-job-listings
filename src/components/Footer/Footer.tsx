import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.footer`
  font-size: 11px;
  text-align: center;

  & a {
    color: hsl(228, 45%, 44%);
  }
`;

function Footer() {
  return (
    <Wrapper>
      Challenge by{' '}
      <a
        href='https://www.frontendmentor.io?ref=challenge'
        target='_blank'
        rel='noreferrer'
      >
        Frontend Mentor
      </a>
      Coded by <a href='https://www.abizmo.dev'>Abi√°n Izquierdo</a>.
    </Wrapper>
  );
}

export default Footer;
