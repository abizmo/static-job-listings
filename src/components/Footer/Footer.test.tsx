import { render, screen } from '@testing-library/react';
import React from 'react';
import Footer from './Footer';

test('render and match snapshot', () => {
  const { asFragment } = render(<Footer />);

  expect(asFragment()).toMatchSnapshot();
});
test('has to links FrontendMentor.io and abizmo.dev', () => {
  render(<Footer />);
  const frontedMentorLink = screen.getByText(/frontend mentor/i);
  expect(frontedMentorLink).toBeInTheDocument();
  expect(frontedMentorLink).toHaveAttribute(
    'href',
    'https://www.frontendmentor.io/?ref=challenge'
  );

  const abizmoLink = screen.getByText(/abi√°n izquierdo/i);
  expect(abizmoLink).toBeInTheDocument();
  expect(abizmoLink).toHaveAttribute('href', 'https://www.abizmo.dev');
});
