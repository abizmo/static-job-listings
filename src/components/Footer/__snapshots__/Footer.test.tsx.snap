// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-bczRLJ bLquij"
  >
    Challenge by 
    <a
      href="https://www.frontendmentor.io/?ref=challenge"
      rel="noreferrer"
      target="_blank"
    >
      Frontend Mentor
    </a>
    Coded by 
    <a
      href="https://www.abizmo.dev"
    >
      Abi√°n Izquierdo
    </a>
    .
  </footer>
</DocumentFragment>
`;
